src/EditorFeatures/CSharpTest/ReverseForStatement/ReverseForStatementTests.cs

```diff
@@ -0,0 +1,368 @@
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; j < args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutVariableReferencedInIncrementor()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; j++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutVariableInitializer()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i; i < args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithMismatchedConditionAndIncrementor1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; i--)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithMismatchedConditionAndIncrementor2()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i >= args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestPostIncrement1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; i++)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(string[] args)
+    {
+        for (int i = args.Length - 1; i >= 0; i--)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestIncrementPreIncrement()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; ++i)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(string[] args)
+    {
+        for (int i = args.Length - 1; i >= 0; --i)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestIncrementAddAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; i += 1)
```

Comment:
Should we care about `i = i + 1`?

Reply:
nah.  it can wait for someone to ask for it.